include/deal.II/fe/fe.h

```diff
@@ -229,15 +260,23 @@ namespace hp
  * the refined lines, since these must be mapped one-to-one to the appropriate
  * dofs of the vertices of the unrefined line.
  *
- * It should be noted that it is not possible to distribute a constrained
- * degree of freedom to other degrees of freedom which are themselves
- * constrained. Only one level of indirection is allowed. It is not known at
- * the time of this writing whether this is a constraint itself.
+ * Through this construction, the degrees of freedom on the child faces are
+ * constrained to the degrees of freedom on the parent face. The information
+ * so provided is typically consumed by the
+ * DoFTools::make_hanging_node_constraints() function.
  *
+ * @note The hanging node constraints described by these matrices is only
+ *   relevant to the case where the same finite element space is used on
+ *   neighboring (but differently refined) cells. The case that the finite
+ *   element spaces on different sides of a face are different, i.e.,
+ *   the $hp$ case (see @ref hp "hp finite element support") is handled
+ *   by separate functions. See the FiniteElement::get_face_interpolation_matrix()
+ *   and FiniteElement::get_subface_interpolation_matrix() functions.
```

Comment:
A question not related to this patch but rather the content in this part of the documentation: We currently still support the "old" way of querying the interface_constraints objects from a finite element via the fe.constraints() method. Does anybody know how many elements still use the old path DoFTools::internal::make_oldstyle_hanging_node_constraints (i.e., which have hp_constraints_are_implemented()==false) rather than the hp variant? If it is not too many, eliminating the old variant may simplify things.

I am most familiar with FE_Q where we go to great lengths to define the interface_constraints matrices alongside the get_subface_interpolation_matrix() functions but the two should really do the same thing as far as I can tell.